File: server/Models/Cult.cs
namespace insta_cult.Models;

public class Cult : RepoItem<int>
{
  // ID
  public string Name { get; set; }
  public double Fee { get; set; }
  public string Description { get; set; }
  public string CoverImg { get; set; }
  public string LeaderId { get; set; }
  public Profile Leader { get; set; }
}

File: client/src/models/Cult.js
import { DatabaseItem } from "./DatabaseItem"
import { Profile } from "./Profile.js"

export class Cult extends DatabaseItem {
  constructor(data) {
    super(data)
    // NOTE these 3 properties are now inherited
    // this.id = data.id
    // this.createdAt = new Date(data.createdAt)
    // this.updatedAt = new Date(data.updatedAt)

    this.name = data.name
    this.fee = data.fee
    this.description = data.description
    this.coverImg = data.coverImg
    this.leaderId = data.leaderId
    this.leader = new Profile(data.leader)
  }
}